scripts/install.tpl.sh

```diff
@@ -41,13 +41,32 @@ function kubectl_wait_rollout() {
     kubectl rollout status --namespace="${namespace}" "${runtime}" --timeout="${timeout}"
 }
 
+function kubectl_wait_for_query() {
+    manifest=$1
+    query=$2
+    timeout=$3
+    start_time=$(date +%s)
+    while true; do
+        val=$(kubectl get "${manifest}" -o jsonpath="${query}" 2>/dev/null || echo "")
+        if [[ -n "${val}" ]]; then
+            echo "${manifest} has ${query}."
+            break
+        fi
+        if [[ $(( $(date +%s) - start_time )) -ge ${timeout} ]]; then
+            echo "Timed out waiting for ${manifest} to have ${query}."
+            exit 1
+        fi
+        sleep 5
+    done
+}
+
```

Comment:
Could this thing be replaced with something along the lines of:
```
kubectl wait validatingwebhookconfigurations/cert-manager-webhook --for=jsonpath='{.webhooks[0].clientConfig.caBundle}' --timeout=60s
```

Reply:
**This is the current approach, but unfortunately, it doesn’t work for our use case.** The kubectl wait command doesn’t handle this situation well. You can test it out by just:


- run kind create cluster
- Then, run the install script  https://operator-framework.github.io/operator-controller/getting-started/olmv1_getting_started/ OR following the steps; https://github.com/operator-framework/operator-controller/pull/1430. You will see the error described in the description.

My Assumption here is: The issue seems to arise because kubectl wait expects JSONPath conditions related to `.status` fields (such as `readyReplicas`), which are applicable for deployments but not for webhook configurations. Since `mutatingwebhookconfigurations` and `validatingwebhookconfigurations` lack these `.status` fields, kubectl wait may fail when it tries to monitor non-existent replica-related conditions. Additionally, it’s possible this issue is linked to cert-manager’s recommendation to use 3 replicas in production environments since we see `error: jsonpath wait format must be --for=jsonpath='{.status.readyReplicas}'=3`, which may indirectly affect how readiness is handled.